ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/IcebergTableMaintainer.java

```diff
@@ -485,7 +485,8 @@ private static int deleteInvalidFilesInFs(
       if (p.isDirectory()) {
         int deleted = deleteInvalidFilesInFs(fio, p.location(), lastTime, excludes);
         deleteCount += deleted;
-        if (!p.location().endsWith(METADATA_FOLDER_NAME)
+        if (fio.exists(p.location())
```

Comment:
Do we need extract the logic to a function? seems the logic here is a bit complicated now.

Reply:
Makes sense, I think we could catch the exception in `clearInternalTableContentsFiles` and print the WRAN log to skip the unexpected error 😶

Reply:
> Do we need extract the logic to a function? seems the logic here is a bit complicated now.

This logic would be refactored (moved to optimizer) by process api, we could do this later